usaspending_api/search/v2/views/search.py

```diff
@@ -145,7 +154,14 @@ def post(self, request):
         """Return all budget function/subfunction titles matching the provided search text"""
         # TODO: check logic in name_dict[x]["aggregated_amount"] statements
 
-        json_request = request.data
+        models = [
+            {'name': 'category', 'key': 'category', 'type': 'enum',
+                'enum_values': ["awarding_agency", "funding_agency", "recipient", "cfda_programs", "industry_codes"],
+                'optional': False}
+        ]
+        models.extend(copy.deepcopy(AWARD_FILTER))
+        models.extend(copy.deepcopy(PAGINATION))
+        json_request = TinyShield(models).block(request.data)
         category = json_request.get("category", None)
         scope = json_request.get("scope", None)
         filters = json_request.get("filters", None)
```

Comment:
(lines 168-169) `limit` and `page` default values should have their defaults applied in the TinyShield models

Also, for non-optional fields (for example, `category`) TinyShield should give confidence that those values exist, so instead of category = json_request.get("category", None) It could just be used as json_request["category"] in the code.